pennylane/drawer/representation_resolver.py

```diff
@@ -331,6 +338,32 @@ def operator_representation(self, op, wire):
 
             return (" " + self.charset.OTIMES + " ").join(constituent_representations)
 
+        if isinstance(op, qml.tape.QuantumTape):
+            if op in self.tape_cache:
+                idx = list(self.tape_cache).index(op) + self.tape_offset
+            else:
+                from .circuit_drawer import CircuitDrawer
+
+                grid, obs = op.graph.greedy_layers()
+                charset_string = next((k for k, v in CHARSETS.items() if v is self.charset), None)
```

Comment:
How come we'd need a generator solution, would this line correspond to something like this?
```python
charset_string = self.charset if self.charset in CHARSETS else None
```

Reply:
Ah originally I was using tape.draw() which only accepts a string descriptor of the charset (self.charset stores the charset class), but since I'm using CircuitDrawer directly now this can be removed. The generator construct is just one way to reverse a dictionary mapping (CHARSETS is a dictionary of {string: class}).